source/Options.cpp

```diff
@@ -147,336 +147,164 @@ Options::Options(
-  options.add_options()(
-      "allow-via-cast-feature",
-      program_options::value<std::vector<std::string>>(),
-      "Compute only these via-cast features. Specified as the full type name, e.g. Ljava/lang/Object;. Multiple inputs allowed. Use --emit-all-via-cast-features to allow everything.");
-
-  options.add_options()(
-      "log-method",
-      program_options::value<std::vector<std::string>>()->multitoken(),
-      "Enable logging for the given methods.");
-  options.add_options()(
-      "log-method-types",
-      program_options::value<std::vector<std::string>>()->multitoken(),
-      "Enable logging of types for the given methods.");
-  options.add_options()(
-      "dump-class-hierarchies",
-      "Dump the class hierarchies in `class_hierarchies.json`.");
-  options.add_options()(
-      "dump-class-intervals",
-      "Dump the class intervals in `class_intervals.json`. For test/debug only.");
-  options.add_options()(
-      "dump-overrides", "Dump the override graph in `overrides.json`.");
-  options.add_options()(
-      "dump-call-graph", "Dump the call graph in `call_graph.json`.");
-  options.add_options()(
-      "dump-dependencies", "Dump the dependency graph in `dependencies.json`.");
-  options.add_options()(
-      "dump-methods", "Dump the list of method signatures in `methods.json`.");
-  options.add_options()(
-      "dump-coverage-info",
-      "Dump the file coverage info into `file_coverage.txt` and rule coverage info into `rule_coverage.json`.");
-
-  options.add_options()(
-      "job-id",
-      program_options::value<std::string>(),
-      "Identifier for the current analysis run.");
-  options.add_options()(
-      "metarun-id",
-      program_options::value<std::string>(),
-      "Identifier for a group of analysis runs.");
-  options.add_options()(
-      "enable-cross-component-analysis",
-      "Compute taint flows across Android components.");
-  options.add_options()(
-      "enable-class-intervals",
-      "Compute and apply class intervals for improved precision.");
-  options.add_options()(
-      "always-export-origins",
-      "Export origin information for all frames instead of only leaves. Used for debugging.");
-  options.add_options()(
-      "propagate-across-arguments",
-      "Enable taint propagation across object type arguments. By default, taint propagation is only tracked for return values and the `this` argument. This enables taint propagation across method invocations for all other object type arguments as well.");
-  options.add_options()(
-      "heuristics",
-      program_options::value<std::string>(),
-      "Path to JSON configuration file which specifies heuristics parameters to use during the analysis. See the documentation for available heuristics parameters.");
+
+std::unique_ptr<Options> Options::options_from_json_file(const std::string& options_json_path){
```

Comment:
nitpick: would remove the additional `options_` from the name.
also: we could take a `std::filesystem::path` to indicate clearly that we expect a path.
```suggestion
std::unique_ptr<Options> Options::from_json_file(const std::string& options_json_path){
```